src/Components/Endpoints/src/Builder/RazorComponentEndpointDataSource.cs

```diff
@@ -172,6 +177,78 @@ private void UpdateEndpoints()
         }
     }
 
+    private void AddBlazorWebEndpoints(List<Endpoint> endpoints)
+    {
+        List<EndpointBuilder> blazorWebEndpoints = [
+            ..GetBlazorWebJsEndpoint(_endpointRouteBuilder),
+            OpaqueRedirection.GetBlazorOpaqueRedirectionEndpoint()];
+
+        foreach (var endpoint in blazorWebEndpoints)
+        {
+            foreach (var convention in _conventions)
+            {
+                convention(endpoint);
+            }
+
+            foreach (var convention in _finallyConventions)
```

Comment:
`RazorComponentEndpointDataSource` ought to implement `GetGroupedEndpoints` and call `_groupConventions` first and `_groupFinallyConventions` last like `RouteEndpointDataSource`, `ControllerActionEndpointDataSource`, and `PageActionEndpointDataSource` do. 

https://github.com/dotnet/aspnetcore/blob/876e36db306a4236bcdd60528305e4acd736aa2d/src/Http/Routing/src/RouteEndpointDataSource.cs#L195-L202

https://github.com/dotnet/aspnetcore/blob/876e36db306a4236bcdd60528305e4acd736aa2d/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs#L346-L350

The default virtual implementation of `GetGroupedEndpoints` provides an approximation of this, but it's not as good because group conventions are forced to run after the `_conventions` and `_finallyConventions` associated with `MapRazorComponent`. It might not be super relevant for these endpoints, but I know `WithOpenApi` looks for already-added metadata in its convention callbacks. The default implementation also prevents a group from injecting any "inner" [endpoint filters](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/min-api-filters) via a group.

https://github.com/dotnet/aspnetcore/blob/5c2dce53828a4435326ae0bce1698a3b8e4824a8/src/Http/Routing/src/EndpointDataSource.cs#L67-L72

These are pretty niche problems. We tried to make the default virtual implementation of `GetGroupedEndpoints` as good as possible, but we might as well make it right.

Reply:
@halter73 I'm going to consider adding support for that outside of scope for this change. I've filed https://github.com/dotnet/aspnetcore/issues/57120 to track it separately, since we are planning on patching this, I want to keep this PR small.

We can discuss the other one separately, as we haven't seen it used in conjunction with `MapGroup` in the wild I think in a way that causes problems.